src/main/java/teammates/ui/webapi/GetInstructorsAction.java

```diff
@@ -28,69 +31,147 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
+                gateKeeper.verifyAccessible(instructor, course);
+            } else {
+                throw new InvalidHttpParameterException("unknown intent");
+            }
         } else {
-            throw new InvalidHttpParameterException("unknown intent");
-        }
+            CourseAttributes course = logic.getCourse(courseId);
+            if (course == null) {
+                throw new EntityNotFoundException("course not found");
+            }
 
+            String intentStr = getRequestParamValue(Const.ParamsNames.INTENT);
+            if (intentStr == null) {
+                // get partial details of instructors with information hiding
+                // student should belong to the course
+                StudentAttributes student = logic.getStudentForGoogleId(courseId, userInfo.getId());
+                gateKeeper.verifyAccessible(student, course);
+            } else if (intentStr.equals(Intent.FULL_DETAIL.toString())) {
+                // get all instructors of a course without information hiding
+                // this need instructor privileges
+                InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
+                gateKeeper.verifyAccessible(instructor, course);
+            } else {
+                throw new InvalidHttpParameterException("unknown intent");
+            }
+        }
     }
 
     @Override
     public JsonResult execute() {
         String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
-        List<InstructorAttributes> instructorsOfCourse = logic.getInstructorsForCourse(courseId);
-
+        String intentStr = getRequestParamValue(Const.ParamsNames.INTENT);
         InstructorsData data;
 
-        String intentStr = getRequestParamValue(Const.ParamsNames.INTENT);
-        if (intentStr == null) {
-            instructorsOfCourse =
-                    instructorsOfCourse.stream()
-                            .filter(InstructorAttributes::isDisplayedToStudents)
-                            .collect(Collectors.toList());
-            data = new InstructorsData(instructorsOfCourse);
-
-            // hide information
-            data.getInstructors().forEach(i -> {
-                i.setGoogleId(null);
-                i.setJoinState(null);
-                i.setIsDisplayedToStudents(null);
-                i.setRole(null);
-            });
-        } else if (intentStr.equals(Intent.FULL_DETAIL.toString())) {
-            // get all instructors of a course without information hiding
-            // adds googleId if caller is admin or has the appropriate privilege to modify instructor
-            if (userInfo.isAdmin || logic.getInstructorForGoogleId(courseId, userInfo.getId()).getPrivileges()
-                    .isAllowedForPrivilege(Const.InstructorPermissions.CAN_MODIFY_INSTRUCTOR)) {
+        if (isCourseMigrated(courseId)) {
```

Comment:
if (!isCourseMigrated(courseId)){
...
}

// No need else here.

This way it's easier to remove the old code without mistakes later.